### ISSUE TITLE
The "restrict access" options are not visible

### ISSUE THREAD

**Issue by gonzalorti** [bozoh/moodle-mod_simplecertificate#208]
Hello,
Congratulations for your plugin.
I have installed it for the first time. I looked at the video. Everything was very easy but... after the "Common module settings" tag I can't see any "Restrict access" nor "Activity completion" set of options.

My Moodle version is 3.4.

My lenguage is Spanish. I attach an image.

Thanks
![captura](https://user-images.githubusercontent.com/43077241/45239563-4a452f80-b2ab-11e8-978d-7b715eadc0a5.JPG)

**Comment by bozoh** [comment]
it's is enabled in moodle system and in the course settings ?

**Comment by gonzalorti** [comment]
Hello, Alexandre.

Indeed, the "Activity completion" was not activated. I have activated it and now it appears in the menu, but not "Restrict Access".
![image](https://user-images.githubusercontent.com/43077241/45266689-7f28c200-b424-11e8-935b-2beea87c7d8e.png)

**Comment by bozoh** [comment]
try to re add the activity ....

**Comment by gonzalorti** [comment]
Yes, I did it. But it's the same.

**Comment by bozoh** [comment]
Restriction access is part of moodle's framework,  is don't  code anything about that, try to add a restriction access to a 'page',  or other activities in the same course.

This plugin have automatic test which test those features in many places,  so i don't think it's have a problem on it.

and remember you must enable in moodle site wide config. AND in the course settings,  course created before enabling restriction in site wide config will have restriction access disabled by default.

More help i think you will find in the plugins forum:
https://moodle.org/mod/forum/discuss.php?d=222296

![20-i-press-save-and-return-to-course-](https://user-images.githubusercontent.com/2317639/45316937-ea4ec300-b50e-11e8-8674-4fee87c0e5e4.png)

**Comment by gonzalorti** [comment]
Thaks, very much, Alexandre. It was that. I succeded to habilitate this option in the site, and I could see the Restrict Access options.

I take the opportunity to suggest a possible improvement. In these options I can chose to restrict access until an selected activity is marked as completed. But I can't find a way **to restrict access until all the activities with completion are completed**. For me, it should be very useful, as this is a very logical requirement. To get this, each time I create a new activity in the course, I have to add it in the certificate restrictions.

Regards,

**Comment by bozoh** [comment]
i think it not in restrict access,  i think it is in completion options

**Comment by gonzalorti** [comment]
I understand that the options for "restrict access" are offered by Moodle and are not specific of your plugin. Anyway, if you can do something to improve them it would be welcome.

In fact, what we require to produce the certificate is that the course has been completed, and for that the administrator of the course set the conditions (cfr. https://docs.moodle.org/35/en/Course_completion_settings): a global grade obtained, the completion of all activities, an attendances percentage of some online or presencial sessions, etc. Or a combination of several of these.

And I see that moodle has criteria that the administrator can set to declare a course completed.

Alas, the restrict access options don't foresee the possibility to require that the course is completed, as they are conceived for activities INTO the course.

It should be perfect for the Simple Certificate plugin to be able to restrict its access until the course is completed.

Regards

**Comment by gonzalorti** [comment]
No, it's the first time I use the completion settings in my courses.
